lib/ui/platform_dispatcher.dart

```diff
@@ -1733,6 +1733,42 @@ enum AppLifecycleState {
   detached,
 }
 
+/// The possible responses to a request to exit the application.
+///
+/// The request is typically responded to by a [WidgetsBindingObserver].
+// TODO(gspencergoog): Insert references here to AppLifecycleListener and to the
+// actual function called on WidgetsBindingObserver once those have landed in
+// the framework.
+enum AppExitResponse {
+  /// Exiting the application can proceed.
+  exit,
+  /// Cancel the exit: do not exit the application.
+  cancel,
+}
+
+/// The type of application exit to perform when calling
+/// `ServicesBinding.exitApplication`.
+// TODO(gspencergoog): Insert references here to ServicesBinding.exitApplication
+// that has landed in the framework.
+enum AppExitType {
+  /// Requests that the application start an orderly exit, sending a request
```

Comment:
```suggestion
  /// Requests that the application starts an orderly exit, sending a request
```

Reply:
I think `start` is correct here since it's the subjunctive 🤓. 

It would be correct to write "Requests the the presenter put on his pants" rather than "Requests that the presenter puts on his pants".

Reply:
![](https://media.tenor.com/SMxNmHW-mf4AAAAd/spy-the-office.gif)